vlsv_reader_parallel.cpp

```diff
@@ -113,7 +116,8 @@ namespace vlsv {
       auto first = multiReadUnits.begin();
       auto last  = multiReadUnits.begin();
       for (auto it=multiReadUnits.begin(); it!=multiReadUnits.end(); ++it) {
-         if (inputBytesize + it->amount*arrayOpen.dataSize > getMaxBytesPerRead()) {
+         if ( (inputBytesize + it->amount*arrayOpen.dataSize > getMaxBytesPerRead()) ||
```

Comment:
To tag it here: I believe this logic is flawed and as at least in Vlasiator we never did proper multiReads until this ill-fated attempt we're working on right now, we never noticed that this logic is flawed. Seems to work in *some* cases, *maybe*, or it only bit us in the specific case of having an advanced state where the bug became a lot more obvious.

Reply:
After discussion: This loop does kinda-correctly construct a list with one copy of each multiReadUnit, but it also ends up plonking a null read at the beginning of the list. Perhaps there's been different behaviour of this in different MPI versions? Anyhow, worth cleaning up.